app/models/project_media.rb

```diff
@@ -32,6 +32,34 @@ class ProjectMedia < ActiveRecord::Base
                   if: proc { |pm| !pm.skip_notifications },
                   data: proc { |pm| pm.media.as_json.merge(class_name: pm.report_type).to_json }
 
+  def relationship_source(relationship_type=Relationship.default_type)
+    Relationship.where(target_id: self.id, relationship_type: relationship_type.to_yaml).last || self
+  end
+
+  def is_claim?
+    self.media_type == "Claim"
+  end
+
+  def is_link?
+    self.media_type == "Link"
+  end
+
+  def is_uploaded_image?
+    self.media_type == "UploadedImage"
+  end
+
+  def is_blank?
+    self.media_type == "Blank"
+  end
```

Comment:
These methods need to be fixed: `media_type` is a setter for the `ProjectMedia` model which is persisted only in memory while the item is being created, so it will be `nil` for database-persisted data. Need to replace `media_type` by `media.type` in all these methods.

Reply:
Ah -good catch - thank you!